[rpc] use annotation_str for RRef type serialization

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39933 [rpc] fix RRef alias annotation
* **#39932 [rpc] use annotation_str for RRef type serialization**

This PR make RRef fork to use the jit type annotation_str recently introduce in
https://github.com/pytorch/pytorch/pull/39544 to allow consistent
serialization type str format, and fix the case when dict->str() format
not match the type resolver.

Differential Revision: [D22015427](https://our.internmc.facebook.com/intern/diff/D22015427)

wanchaol committed:
[rpc] use annotation_str for RRef type serialization

This PR make RRef fork to use the jit type annotation_str recently introduce in
https://github.com/pytorch/pytorch/pull/39544 to allow consistent
serialization type str format, and fix the case when dict->str() format
not match the type resolver.

[ghstack-poisoned]

mrshenli reviewed (approved):
Thanks for fixing promptly!

wanchaol committed:
Update on "[rpc] use annotation_str for RRef type serialization"


This PR make RRef fork to use the jit type annotation_str recently introduce in
https://github.com/pytorch/pytorch/pull/39544 to allow consistent
serialization type str format, and fix the case when dict->str() format
not match the type resolver.

Differential Revision: [D22015427](https://our.internmc.facebook.com/intern/diff/D22015427)

[ghstack-poisoned]
